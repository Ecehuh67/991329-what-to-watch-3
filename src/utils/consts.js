export const DEFAULT_GENRE = `All genres`;
export const DEFAULT_SHOWED_FILMS = 8;
export const DEFAULT_GENRES_AMOUNT = 9;
export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  MAIN: `/`,
  SIGN_IN: `/login`,
  MY_LIST: `/mylist`,
  FILM: (id) => `/films/${id}`,
  PLAYER: (id) => `/player/${id}`,
  REVIEW: (id) => `/films/${id}/review`
}

export const EstimateValue = {
  3: "Bad",
  5: "Normal",
  8: "Good",
  10: "Very good",
  11: "Awesome"
};
