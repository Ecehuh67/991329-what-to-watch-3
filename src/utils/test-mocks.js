export const movies = [
  {
    title: `The Lord of the Rings`,
    image: `img/the-lord-of-the-rings.jpg`
  },
  {
    title: `The Ghostbusters`,
    image: `img/the-ghostbusters.jpg`
  },
  {
    title: `The Mortal Combat`,
    image: `img/the-mortal-combat.jpg`
  }
];
