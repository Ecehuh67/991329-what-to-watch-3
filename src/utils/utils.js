const MONTHS = {
  0: `January`,
  1: `February`,
  2: `March`,
  3: `April`,
  4: `May`,
  5: `June`,
  6: `Juily`,
  7: `Augest`,
  8: `September`,
  9: `October`,
  10: `November`,
  11: `December`
};

export const getRandomNumber = (max) => {
  return 1 + Math.floor(Math.random() * max);
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const convertTime = (duration) => {
  const hours = Math.floor(duration / 60);
  const minutes = Math.round((duration / 60 - hours) * 60);

  return (
    `${hours}h ${minutes}m`
  );
};

export const convertTimeToProgressBar = (duration) => {
  const hours = Math.floor(duration / 3600);
  const minutes = Math.floor((duration - hours * 3600) / 60);
  const seconds = Math.round(duration - hours * 3600 - minutes * 60);
  return (
    `${hours}:${minutes}:${seconds < 10 ? `0${seconds}` : seconds}`
  );
};

export const getCommentDate = (date) => {
  const month = date.getMonth();
  const day = date.getDay();
  const year = date.getFullYear();

  return `${MONTHS[month]} ${day}, ${year}`;
};

export const getAttrDate = (date) => {
  const month = date.getMonth();
  const day = date.getDay();
  const year = date.getFullYear();

  return `${year}-${month < 10 ? `0${month}` : month}-${day < 10 ? `0${day}` : day}`;
};

export const getNewData = (newElement, arrayData) => {
  const newIndex = arrayData.findIndex((movie) => movie.id === newElement.id);
  const newData = arrayData.slice(0, newIndex).concat(arrayData.slice(newIndex + 1)).concat(newElement).sort((a, b) => a.id - b.id);

  return newData;
};

export const sortData = (newElement, arrayData) => {
  const newIndex = arrayData.findIndex((movie) => movie.id === newElement.id);
  const oldData = arrayData;

  if (newIndex === -1) {
    oldData.push(newElement);
  } else {
    oldData.splice(newIndex, 1);
  }

  return oldData;
};

export const getRank = (count) => {
  let rank;

  switch (true) {
    case count > 10:
      rank = `Awesome`;
      break;
    case count > 8 && count <= 10:
      rank = `Very good`;
      break;
    case count > 5 && count <= 8:
      rank = `Good`;
      break;
    case count > 3 && count <= 5:
      rank = `Normal`;
      break;
    case count > 0 && count <= 3:
      rank = `Bad`;
      break;
  }

  return rank;
};
