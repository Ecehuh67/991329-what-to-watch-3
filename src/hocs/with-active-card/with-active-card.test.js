import withActiveCard from "./with-active-card";
import {movies} from '../../utils/test-mocks';

const film = movies[1];

const MockComponent = (props) => {
  const {children} = props;

  return (
    <>
      {children}
    </>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withActiveCard(MockComponent);

it(`withActiveCard is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped
      film={film}
      onDataChange={() => {}}
    />
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
