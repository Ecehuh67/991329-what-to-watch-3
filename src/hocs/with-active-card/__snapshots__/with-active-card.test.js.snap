// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withActiveCard is rendered correctly 1`] = `
<img
  alt="Fantastic Beasts: The Crimes of Grindelwald"
  height="175"
  src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
  width="280"
/>
`;
